### PR TITLE
Refactor Right Panel

### PR DESCRIPTION
**Pull request by skyqrose**
Pulls the logic for the right panel out of `LadderPage` so it will be easier to add the notifications drawer to and use on every page.
Supports [Bell icon and opening an empty notification drawer](https://app.asana.com/0/1148853526253426/1192202587402575)

Also:
* Moves the `useRoutes()` fetch to `appStateWrapper` so we only make it once and it's accessible everywhere.
* Adds a `useRoute()` wrapper around that context for the common case of getting one route by id.
* Changes `useVehicleAndRouteForNotification` to `useVehicleForNotification` now that we can get the route from anywhere.
* Moves `useVehicleForNotification` to `appStateWrapper` with a context so `modal.tsx` can see the result, removing the need for setting `state.notificationIsLoading` and `state.notificationIsInvalid`.

Right now, the `RightPanel` is only called by `LadderPage`, but in the future, every page will have its own `RightPanel`.

This PR should have no user-visible changes.

### CODE REVIEW

File: assets/src/components/appStateWrapper.tsx [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
   const socketStatus = useSocket()
+  const routes = useRoutes()
 
   return (
     <StateDispatchProvider state={state} dispatch={dispatch}>
       <SocketProvider socketStatus={socketStatus}>
-        <NotificationsProvider>
-          <App />
-        </NotificationsProvider>
+        <RoutesContext.Provider value={routes}>
```
Comment (phildarnowsky) [inline assets/src/components/appStateWrapper.tsx:25]: @skyqrose for consistency let's define a `RoutesProvider` component in `routes_context.tsx`

File: assets/tests/components/__snapshots__/ladderPage.test.tsx.snap [hunk 2]
```diff
@@ -509,15 +526,15 @@ exports[`LadderPage renders with timepoints 1`] = `
       >
         <li>
           <button
-            className="m-route-picker__route-list-button m-route-picker__route-list-button--unselected"
+            className="m-route-picker__route-list-button m-route-picker__route-list-button--selected"
```
Comment (phildarnowsky) [inline assets/tests/components/__snapshots__/ladderPage.test.tsx.snap:529]: does this change and the similar one below from `unselected` to `selected` represent a possible regression?

File: assets/src/components/rightPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      dispatch(setNotificationIsLoading(false))
+    }
+  }, [vehicleAndRouteForNotification, selectedNotification])
+  const history = useHistory()
+  if (history) {
+    /* istanbul ignore next */
+    history.listen(() => dispatch(setNotification(undefined)))
+  }
+
+  const vehicleOrGhostForVPP:
```
Comment (phildarnowsky) [inline assets/src/components/rightPanel.tsx]: not directly related to the PR but while we're here maybe let's add a comment about the `null`/`undefined` convention for loading vehicles

File: assets/src/components/appStateWrapper.tsx [hunk 4]
```diff
@@ -3,22 +3,32 @@ import { NotificationsProvider } from "../contexts/notificationsContext"
...
+    state.selectedNotification
+  )
+
   return (
     <StateDispatchProvider state={state} dispatch={dispatch}>
       <SocketProvider socketStatus={socketStatus}>
         <RoutesContext.Provider value={routes}>
           <NotificationsProvider>
-            <App />
+            <VehicleAndRouteForNotificationContext.Provider
```
Comment (phildarnowsky) [inline assets/src/components/appStateWrapper.tsx]: likewise let's have a `VehicleAndRouteForNotificationProvider`

**Review by phildarnowsky — APPROVED** [review]
@skyqrose nice work simplifying this stuff, I have only minor feedback.

File: assets/tests/components/__snapshots__/ladderPage.test.tsx.snap (hunk 2 above)
Comment (skyqrose) [reply assets/tests/components/__snapshots__/ladderPage.test.tsx.snap:529]: No.
The test `renders with timepoints` didn't select the routes, so it didn't actually render with timepoints.  [Fixing that test to be as intended](https://github.com/mbta/skate/pull/899/files#diff-41b776228e88467a5a40b857932c4e403705be68b76270e8b50f440ec6abcb3cR65) is the cause of all the changes in this ladderPage snapshot.

**Pull request merged**